src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs

```diff
@@ -320,10 +321,15 @@ public new IntPtr DangerousGetHandle()
 
     internal sealed class SafeFreeCredential_SECURITY : SafeFreeCredentials
     {
+#pragma warning disable 0649
+         // This is used only by SslStream but it is included elsewhere
+         public X509Certificate? LocalCertificate;
+ #pragma warning restore 0649
         public SafeFreeCredential_SECURITY() : base() { }
 
         protected override bool ReleaseHandle()
         {
+            LocalCertificate?.Dispose();
```

Comment:
Same question as on the 7.0 PR

Reply:
Let's keep the disucssion here for both PRs.

I don't see anything in the dispose implementation nor transitively called methods that could throw. The Reset method which is called there is virtual, *but* we explicitly make sure we assign only `X509Certificate2` instances to the field.

Reply:
This is what current `main` does. If we are concern we may wrap it In `try/catch` or simply re-order the code. In either case my preference would be to keep it consistent across all supported branches to make future back ports easier